xdsl/dialects/tosa.py

```diff
@@ -124,31 +126,70 @@ class AddOp(IRDLOperation):
 
     traits = traits_def(
         Pure(),
-        Commutative(),
     )
 
     def verify_(self) -> None:
-        """
-        Verify that the two input tensors are compatible, and that the result type can be constructed. For this,
-        both tensors must have the same rank. They should either have the same number of elements per dim, or
-        if there is only one element it can be broadcast implcitly.
-        """
         t1 = self.input1.type
         t2 = self.input2.type
         t_out = self.output.type
 
         if not are_tosa_broadcastable(t1, t2, t_out):
             raise VerifyException(
-                "'tosa.add' Operand and result tensor shapes are not compatible"
+                f"'{type(self).name}' Operand and result tensor shapes are not compatible"
             )
 
 
+@irdl_op_definition
+class AddOp(ElementwiseBinaryOperation):
+    """
+    Tosa elementwise add operation
+
+    https://mlir.llvm.org/docs/Dialects/TOSA/#tosaadd-mlirtosaaddop
+    """
+
+    name = "tosa.add"
+
+    traits = traits_def(
+        Pure(),
```

Comment:
The traits stack instead of being overridden BTW so you don't need to repeat them below.
```suggestion
```